packages/typespec-client-generator-core/test/public-utils.test.ts

```diff
@@ -921,7 +923,7 @@ describe("typespec-client-generator-core: public-utils", () => {
         const models = runner.context.experimental_sdkPackage.models;
         strictEqual(models.length, 1);
         strictEqual(models[0].name, "TestRequest");
-        strictEqual(models[0].crossLanguageDefinitionId, "TestService.TestRequest");
+        strictEqual(models[0].crossLanguageDefinitionId, "TestService.test.Request.anonymous");
```

Comment:
I see a lot of `Service.blah.Foo.anonymous` tests in here, what about one for:
```
model Foo {
  props: {
    name: string;
  }
}
```
The cross language id for that name would be `Foo.props.anonymous.name`.

Reply:
we do have quite a few of those, one example would be line 1541 in the same test file